packages/solana-connector/src/utils/Predicate.ts

```diff
@@ -0,0 +1,115 @@
+import { Keypair } from '@solana/web3.js';
+import bs58 from 'bs58';
+import {
+  Address,
+  type BN,
+  type InputValue,
+  type JsonAbi,
+  Predicate,
+  type Provider,
+  ScriptTransactionRequest,
+  ZeroBytes32,
+  arrayify,
+  bn,
+  getPredicateRoot,
+  hexlify,
+  sha256,
+} from 'fuels';
+import memoize from 'memoizee';
+import nacl from 'tweetnacl';
+import type { PredicateConfig } from '../types';
+
+export class PredicateAccount {
+  private abi: JsonAbi;
+  private bytecode: Uint8Array;
+
+  constructor({ abi, bytecode }: PredicateConfig) {
+    this.abi = abi;
+    this.bytecode = bytecode;
+  }
+
+  getPredicateAddress = memoize((svmAddress: string): string => {
+    const configurable = {
+      SIGNER: hexlify(bs58.decode(svmAddress)),
```

Comment:
We could move this part to be a separate function on the class `getAddress()`. In this way, once you start refactoring other predicates, you can move this to a common package and extend the class overriding only the `getAddress`.

Reply:
Same as the comment above